### ISSUE TITLE
Applying right-click "Filter" options on a table creates invalid Z

### ISSUE THREAD

**Issue by philrz** [brimsec/brim#1360]
A community user pointed out:

> In the Queries of "Activity Overiew",It' is error if you select "Filter = value" or other.

Indeed, this is true of aggregation tables in general, but the new Query Library places such tables more front-and-center in with users and hence more likely to notice this.

Here's an example repro with GA Brim `v0.22.0`.

https://user-images.githubusercontent.com/5934157/104854905-328c0600-58be-11eb-80ac-bec7d5385571.mov

**Comment by jameskerr** [comment]
Hey @philrz will you help find the desired behavior here? Should I disable the filter=/!=value options when there is an aggregation present? 

The logic now is "place this filter before the first pipe you see". I can make it instead "place this filter at the start of the string.

Not sure what the best fix for this is.

**Comment by philrz** [comment]
@jameskerr: I agree that it's not obvious. I was prepared for the likelihood that it might just have to be disabled. Now that you've explained the current logic, I could see some arguments for keeping it in play, but it still seems hazardous.

In defense of the current logic, I can see it's handy even when an aggregation is present, as long as the Z pipeline starts with a search.

https://user-images.githubusercontent.com/5934157/105255138-e0eeb000-5b37-11eb-9b71-ccefa0d4e037.mov

On the other hand, since a Z pipeline can start with any processor or aggregation, here we see it break.

https://user-images.githubusercontent.com/5934157/105255218-04b1f600-5b38-11eb-9695-76c0697d3146.mov

When you say you could instead "place this filter at the start of the string", are you saying you'd prepend not only the filter terms but also an additional pipe operator, like this?

![image](https://user-images.githubusercontent.com/5934157/105255474-7c802080-5b38-11eb-98fa-e4a70b46d450.png)

If so, that might be convenient, but then it seems like you'd need to have knowledge of whether what you're prepending before is a search or a processor/aggregation, since (at least currently) you can't pipe the output from one search to yet another search.

![image](https://user-images.githubusercontent.com/5934157/105255649-d8e34000-5b38-11eb-9c09-62cbbab8bf07.png)

If this seems too complex to deal with, I'd be fine with just disabling when an aggregation is present. We could also talk about it with the wider team tomorrow morning.

**Comment by philrz** [comment]
We discussed this as a group today. It sounds like it's technically feasible for the app to look into the AST to determine the nature of the first element in the Z query. If the first element found is a search, then the filter will just be prepended so it becomes an additional part (boolean `and`) of the existing search. If it's not a search (i.e. it's some processor or aggregate function) then the filter will be prepended along with a `|` operator.
